`class-methods-use-this` ESLint 규칙 무시

somnisomni opened the pull request:
TS 클래스 스타일 Vue 컴포넌트 작성 시, 해당 ESLint 규칙을 따르게 되면 한정자(static) 때문에 일부 클래스 멤버를 정상적으로 사용할 수 없게 됩니다.
규칙 문서 : https://eslint.org/docs/rules/class-methods-use-this

예시로,
```typescript
export default class ExampleComponent extends Vue {
  get testGetter(): string {
    return "some data";
  }
}
```
는 정상적으로 작동(`<template>` 내에서 바인드하여 사용하여도 문제 없음)하는 컴포넌트 클래스 코드입니다.
하지만 해당 규칙은 `testGetter()` 메서드 내에서 자신의 부모 클래스를 참조하는 `this` 키워드가 사용되지 않았기 때문에 클래스 한정 멤버일 필요가 없다고 판단, static으로 작성할 것을 강제합니다.

규칙을 따라 수정한 코드(`testGetter()` 부분만),
```typescript
static get testGetter(): string {
  return "some data";
}
```
는 `<template>` 내에서 사용하려고 시도하는 경우 동작하지 않음을 확인했습니다.

이런 Vue의 특성으로 인해 해당 규칙을 무시하여야 한다는 결론을 냈습니다.

somnisomni committed:
Ignore `class-methods-use-this` eslint rule

TS 클래스 스타일 Vue 컴포넌트 작성 시, 해당 ESLint 규칙을 따르게 되면
한정자(static) 때문에 일부 컴포넌트 클래스 멤버를 정상적으로
사용할 수 없게 됩니다.
규칙 문서 : https://eslint.org/docs/rules/class-methods-use-this

예시로,
`export default class ExampleComponent extends Vue {
  get testGetter(): string {
    return "some data";
  }
}`
는 정상적으로 작동(`<template>` 내에서 바인드하여 사용하여도 문제 없음)하는
컴포넌트 클래스 코드입니다. 하지만 해당 규칙은 `testGetter()` 메서드 내에서
자신의 부모 클래스를 참조하는 `this` 키워드가 사용되지 않았기 때문에
클래스 한정 멤버일 필요가 없다고 판단, static으로 작성할 것을 강제합니다.

규칙을 따라 수정한 코드(`testGetter()` 부분만),
`static get testGetter(): string {
  return "some data";
}`
는 `<template>` 내에서 사용하려고 시도하는 경우 동작하지 않음을 확인했습니다.

이런 Vue의 특성으로 인해 해당 규칙을 무시하여야 한다는 결론을 냈습니다.

somnisomni committed:
Remove unused `require`

woosukji commented on Web/.eslintrc.js:
```diff
@@ -15,7 +15,7 @@ module.exports = {
     "lines-between-class-members": "off",
+    "class-methods-use-this": "off",
     quotes: ["error", "double"],
-    "@typescript-eslint/no-var-requires": 0,
```
이 부분을 지울 이유는 없을 것 같아요.
이게 `require` 을 `var/let/const`로 받는 걸 막고 `import` 로만 받으라는 조건이어서,
[PR#6](https://github.com/osamhack2021/AI_WEB_POOL_YD/pull/6/files)에 올리신 이 라인같은 경우도 경고 뜰거라서요
(js 파일에도 같은 rule을 쓰는 것 같던데, 이거 막을 수 없나요?)
https://github.com/osamhack2021/AI_WEB_POOL_YD/blob/7cf482e55bb3cd591ed45cc563f4d0eefe0554a5/Web/scripts/gh-pages-deploy.js#L1

somnisomni replied:
아... 그렇네요...
9017b58cd73d23d5b7078abc7179f9fdce58fd7a 에서 변경된 [vue.config.js](https://github.com/osamhack2021/AI_WEB_POOL_YD/blob/9017b58cd73d23d5b7078abc7179f9fdce58fd7a/Web/vue.config.js)만 보고 *(이 부분 require가 더 이상 필요없어서)* 규칙을 삭제해버렸네요.  
곧바로 다시 추가하겠습니다.

파일별로 규칙을 비활성화하려면 `// eslint-disable` 같은걸 써야될 것 같네요. 이것도 .js 파일들에 포함해서 커밋 올리겠습니다.

somnisomni replied:
@woosukji 말씀하신 부분은 모던 ES6 스타일의 `import`문 사용을 권장하는 규칙이기 때문에 전역적으로 이 규칙을 비활성화하기보단 .js 파일에 규칙이 어겨지는 줄만 ESLint disable 하는걸로 대신했습니다. 91f13bc 리뷰 부탁드립니다!

woosukji reviewed (commented):
OK

somnisomni committed:
Add ESLint disable comment for `.js` files

모듈 로드는 ES6 스타일의 `import`문을 사용하는 것이
권장되므로, `no-var-requires` 규칙을 전역적으로
비활성화하진 않습니다. 대신, 해당 규칙 비활성화가
필요한 자바스크립트 파일 내에 비활성화 주석으로 대신합니다.

woosukji reviewed (approved):
💯
